### DISCUSSION TITLE
Modify Generated Types with Dynamic Property

### DISCUSSION THREAD

**Discussion by tony-cocco** [prisma/prisma#14364] [Q&A] [answered]
Hello All,

I'm currently attempting to add in a dynamic prop to all of my prisma calls in order to handle some auditing in prisma middleware. So for example:

```
await db.post.update({context: { /* stuff */}, where: { id: 1 } });
```

In middleware, I'm able to read the data out of context and do stuff:

```
prisma.$use(async (params: Prisma.MiddlewareParams, next) => {
  const { context } = params.args;
  /* do stuff with context */
  delete params.args.context;
  next(params);
```

My issue is that TypeScript complains:

![image](https://user-images.githubusercontent.com/4268000/179552816-636a176b-a379-4a0c-a3f0-65e50e5d5d8a.png)

I haven't figured out a nice way to amend the type and would prefer to not have to pop @ts-ignores everywhere. Any suggestions? None of my searches have turned up anything helpful.

**Comment by well-balanced** [comment, marked as answer]
I think they're still discussing for the feature

https://github.com/prisma/prisma/issues/6882

**Reply by tony-cocco** [reply]
Thank you for finding this.

**Reply by well-balanced** [reply]
I hope you have some good news :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
